react-app/src/app/components/PublicRoute/index.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+class PublicRoute extends Component {
+  render() {
+    if (this.props.logged && this.props.path === '/login') {
```

Comment:
Notice that this component is pretty similar to `RestrictedRoute`. You could define a single component that contains both logics based on a `isPrivate` or `isPublic` prop.